src/GoTrueClient.ts

```diff
@@ -312,13 +312,17 @@ export default class GoTrueClient {
    * For server-side management, you can disable sessions by passing a JWT through to `auth.api.signOut(JWT: string)`
    */
   async signOut(): Promise<{ error: Error | null }> {
-    this._removeSession()
-    this._notifyAllSubscribers('SIGNED_OUT')
     if (this.currentSession) {
```

Comment:
the original commit is [here](https://github.com/supabase/gotrue-js/commit/3bc79f6b56f4ffae979e167a0fd0b67ce02cfe93)

we can fix it by copy the access_token first.
```
const accessToken = this.currentSession?.access_token
this._removeSession()
this._notifyAllSubscribers('SIGNED_OUT')
if (accessToken) {
   const { error } = await this.api.signOut(accessToken)
   ...
}
```

Reply:
Done.  Look good?